### PR TITLE
Feat(ai): first implementation of the PaletteNet algorithm managed from image-go-nord package

### LINKED ISSUE #157: Create the imagIA section
As per title

### PR DESCRIPTION
**Pull request by TheJoin95**
# Pull Request

Closes #157 

## Description

This is the first implementation of the AI model PaletteNet available from the image-go-nord package.
It's accepting a palette of 6 colors and it's then converting it using the model made in pytorch.

A second phase would convert the pytorch model into javascript model in order to be executed on the client side.
It will be discovered in a different issue.

Documentation will be updated in a follow-up ticket (#163 )

image

image

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] tested locally

It needs to be tested on prod, as we need to update and get the code on our VPS.

**Test Configuration**:
* Firmware version: any
* Hardware: any
* Toolchain: Python, Vue
* IGN Version: 1.1.0

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by TheJoin95** [comment]
This pr has done a long journey but I'm happy to make it online

**Review by Wabri — CHANGES_REQUESTED** [review]
Little changes and I'm in your hands for the css staff. Huge work!

File: src/ign-api/conversion.py [hunk 1]
```diff
@@ -39,12 +39,16 @@ def convert_queue():
 
   if (request.form.get('palette_name') != None):
     palette_name = request.form.get('palette_name')
+  
+  use_model = False
+  if (request.form.get('is_ai') != None):
```
[the remark this answers is not in the document]
Comment (TheJoin95) [reply src/ign-api/conversion.py:44]: Totally and I agree with you

File: src/ign-api/conversion.py [hunk 2]
```diff
@@ -39,12 +39,16 @@ def convert_queue():
...
+    use_model = True
 
   conn.incr('conversion_count', 1)
 
   if is_image(file.filename):
     image = go_nord.open_image(request.files.get('file').stream)
-    job = q.enqueue(f=convert_image, ttl=900, failure_ttl=900, job_timeout='180s', args=(go_nord, image, output_path, request.form.get('b64_output'), response))
+    job = q.enqueue(f=convert_image, ttl=900, failure_ttl=900, job_timeout='180s', args=(go_nord, image, output_path, request.form.get('b64_output'), response, use_model))
```
[the remark this answers is not in the document]
Comment (TheJoin95) [reply src/ign-api/conversion.py:51]: Let's keep it as they are, we might need to do some changes in the near future to this queue logic

File: src/ign-frontend/src/components/Demo.vue [hunk 3]
```diff
@@ -111,18 +122,24 @@ export default Vue.component('Demo', {
...
+        // eslint-disable-next-line
+        this.palette = require(`../assets/${file}`);
+      } catch (e) {
+        // eslint-disable-next-line
+        this.palette = require('../assets/palettes/ai-palettes/6x-palette.json').find((p) => p.name === file);
+      }
     },
     palette(p) {
-      this.selectedColor = p.colors;
+      this.selectedColor = p?.colors;
```
[the remark this answers is not in the document]
Comment (TheJoin95) [reply src/ign-frontend/src/components/Demo.vue:142]: Old code from a WIP commit: there will be always a `colors` key in the palette object.
Thanks for the catch

**Review by Wabri — APPROVED** [review]

**Pull request merged**